server/server.ts

```diff
@@ -107,6 +107,34 @@ app.get('/ping', (_req, res) => {
   res.json('pong');
 });
 
+app.post('/discord/auth', async (req, res) => {
+  const decoded = await validateUserToken(req.body?.uid, req.body?.token);
+  if (!decoded) {
+    return res.status(400).json({ error: 'invalid user token' });
+  }
+  if (!postgres) {
+    return res.sendStatus(500);
+  }
+  try {
+    // users should only have one account max. with a sub role per email,
+    // set email to null to tell background process to clean it up.
+    await postgres?.query('UPDATE discord SET email = null WHERE email = $1', [
```

Comment:
wonder if it's better to have one links table that can support linking different types of accounts in the future by adding more columns?

Reply:
Do you mean a table which has a column for each service's user ID? I think with a table like that we would not have to be matching the email address anymore.

Reply:
we would still be matching the email to find the row, but that row could support multiple services.